### PR TITLE
Add support for iptables in nftables mode.

### PR DESCRIPTION
**Pull request by squeed**
Iptables also has the ability to work in nftables mode, where it is supposed to act like iptables but use the nftables subsystem. Unfortunately, it isn't exactly the same.

The biggest difference is that counter output is iptables-save style, rather than with "-c N N".

Also, improve some tests.

Fixes: #49

### CODE REVIEW

**Comment by squeed** [comment]
@lucab would you mind taking a look?

File: test [hunk 1]
```diff
@@ -45,11 +45,14 @@ split=(${TEST// / })
...
+go test -c -o ${bin} ${COVER} -i ${TEST}
 if [[ -z "$SUDO_PERMITTED" ]]; then
     echo "Test aborted for safety reasons. Please set the SUDO_PERMITTED variable."
     exit 1
 fi
 
-sudo -E bash -c "PATH=\$GOROOT/bin:\$PATH go test ${COVER} $@ ${TEST}"
+sudo -E bash -c "${bin} $@ ${TEST}"
 echo "Success"
+rm ${bin}
```
Comment (lucab) [inline test]: Here and above, please quote variables in `""`: `"${bin}"` etc.

File: iptables/iptables.go [hunk 2]
```diff
@@ -473,3 +490,24 @@ func (ipt *IPTables) existsForOldIptables(table, chain string, rulespec []string
...
+func filterRuleOutput(rule string) string {
+	// work around an output difference in nftables mode where counters
+	// are output in iptables-save format, rather than iptables -S format
+	// The string begins with "[0:0]"
+	//
+	// Fixes #49
+	if groups := counterRegex.FindStringSubmatch(rule); groups != nil {
+		// drop the brackets
+		rule = rule[len(groups[0]):]
+		rule = fmt.Sprintf("%s -c %s %s", rule, groups[1], groups[2])
```
Comment (lucab) [inline iptables/iptables.go]: Nit: I'd personally prefer not shadowing input parameters and local variables, if possible.

**Comment by squeed** [comment]
Updated, and added another fix for a failing testcase; PTAL

**Pull request merged**